apps/web/src/pages/WorkStreams/UpdateWorkStreamPage.tsx

```diff
@@ -0,0 +1,376 @@
+  fragment WorkStreamUpdate on WorkStream {
+    id
+    name {
+      en
+      fr
+      localized
+    }
+    plainLanguageName {
+      en
+      fr
+    }
+    community {
+      id
+      key
+      name {
+        en
+        fr
+      }
+    }
+  }
+`);
+
+interface FormValues {
+  key?: InputMaybe<Scalars["String"]["input"]>;
+  name?: InputMaybe<LocalizedStringInput>;
+  plainLanguageName?: InputMaybe<LocalizedStringInput>;
+  community: string;
+}
+
+const formValuesToSubmitData = (data: FormValues): UpdateWorkStreamInput => {
+  const communityId = data.community;
+  return {
+    name: {
+      en: data.name?.en,
+      fr: data.name?.fr,
+    },
+    plainLanguageName: {
+      en: data.plainLanguageName?.en,
+      fr: data.plainLanguageName?.fr,
+    },
+    community: { connect: communityId },
+  };
+};
+
+interface UpdateWorkStreamProps {
+  query: FragmentType<typeof WorkStreamUpdate_Fragment>;
+  communityOptions: OptGroupOrOption[];
+}
+
+export const UpdateWorkStreamForm = ({
+  query,
+  communityOptions,
+}: UpdateWorkStreamProps) => {
+  const intl = useIntl();
+  const navigate = useNavigate();
+  const paths = useRoutes();
+  const [, executeMutation] = useMutation(UpdateWorkStream_Mutation);
+  const workStream = getFragment(WorkStreamUpdate_Fragment, query);
+
+  const methods = useForm<FormValues>({
+    defaultValues: {
+      name: workStream.name,
+      plainLanguageName: workStream.plainLanguageName,
+      community: workStream.community?.id,
+    },
+  });
+  const { handleSubmit } = methods;
+
+  const handleError = () => {
+    toast.error(
+      intl.formatMessage({
+        defaultMessage: "Error: updating work stream failed",
+        id: "uNKq32",
+        description:
+          "Message displayed to user after work stream fails to get updated.",
+      }),
+    );
+  };
+
+  const onSubmit: SubmitHandler<FormValues> = async (data: FormValues) => {
+    return executeMutation({
+      id: workStream.id,
+      workStream: formValuesToSubmitData(data),
+    })
+      .then(async (result) => {
+        if (result.data?.updateWorkStream) {
+          await navigate(
+            paths.workStreamView(result.data?.updateWorkStream.id),
+          );
+          toast.success(
+            intl.formatMessage({
+              defaultMessage: "Work stream updated successfully!",
+              id: "0cieq3",
+              description:
+                "Message displayed to user after work stream is updated successfully.",
+            }),
+          );
+        } else {
+          handleError();
+        }
+      })
+      .catch(handleError);
+  };
+
+  const navigationCrumbs = useBreadcrumbs({
+    crumbs: [
+      {
+        label: intl.formatMessage(pageTitles.workStreams),
+        url: paths.workStreamTable(),
+      },
+      {
+        label: workStream.name?.localized,
+        url: paths.workStreamView(workStream.id),
+      },
+      ...(workStream.id
```

Comment:
This check should no longer be necessary since it must exist to get to this point